lib/trollop.rb

```diff
@@ -635,169 +801,102 @@ def cloaker(&b)
   end
 end
 
-## The option for each flag
-class Option
```

Comment:
Moving `Option` makes comparison difficult here.

Reply:
I ran into an ordering issue, in that the registry hash references the `Option` subclasses and if I reference (for example) `IntegerOption` before it's defined, then i get a missing constant error.  Another way is to reopen the `Parser` class and add it later, after `Option` subclasses are defined.

Another way would be to have the option subclasses register themselves, e.g.
```ruby 
class IntegerOption
  register :int, :integer, :fixnum
  ....
end
```
instead of
```
  @registry = {
    :fixnum => IntegerOption,
    :int => IntegerOption,
    :integer => IntegerOption,

```

Reply:
I like external registry

Reply:
Ok. So the problem is parse is defined before Options. I’ll dive into the changes a little more deeply